### DISCUSSION TITLE
Qualys Payloads

### DISCUSSION THREAD

**Discussion by Shoganai99** [DefectDojo/django-DefectDojo#10239] [Q&A] [answered]
Hi everyone,

I have an issue in DefectDojo when import a Qualys Web Application Scan, in te XML report from Qualys i see the value of the payload tested by Qualys but in DefectDojo these value is missing and the requests field is incomplete.

There is an example, in the first image you can see the payload in the XML report
1.![1 payload_xml](https://github.com/DefectDojo/django-DefectDojo/assets/59613220/a5c97953-f3a2-4912-98ff-ce915f3d6763)

In the second image you can see one of the requests maded by Qualys whith the previous payload in the XML body tag
2.
![2 request_xml](https://github.com/DefectDojo/django-DefectDojo/assets/59613220/7e4590d7-3307-44d2-87b7-49af971ddaad)

In the third image you can see the same request imported in DefecDojo and the body tag from XML file is missing
3.
![3 request_ddojo](https://github.com/DefectDojo/django-DefectDojo/assets/59613220/d979b1fd-45b1-4637-8be6-780fd27a305e)

In the four imagen you can see that the steps to reproduce drop-down is empty
4.
![4 steps_to_reproduce](https://github.com/DefectDojo/django-DefectDojo/assets/59613220/84679819-2a5a-424e-9f6c-0f66ce740fc2)

I replicate this scenario in 2 different Ddojo versions 2.31.3 and 2.34.3, the first one in LinuxMint and the second in Ubuntu, to install and run the 2 versions i used django with docker compose v2

**Comment by manuel-sommer** [comment, marked as answer]
Hi @Skuller99Dev ,

could you please open up an issue and share a sample file with a short description on what values are exactly missing. Maybe these scans already help: https://github.com/DefectDojo/django-DefectDojo/tree/master/unittests/scans/qualys_webapp. Then, I might be able to help you with a PR to resolve this issue.

**Reply by Shoganai99** [reply]
Hello,

First of all thank you very much for the quick response,
There is an example report, it is one of the modified qualys example reports because the example reports do not have post parameters tested with malicious code, in this example I copy a real request and response but obfuscating key parts like the real web.

As you can see in the XML there is a post parameter with malicious code sent to the web page via POST request, the problem is that in defect dojo, in the request, the post parameter is not shown, so I can't know what is the value of the payload sent to the web.

![1 Request_and_response_in_ddojo](https://github.com/DefectDojo/django-DefectDojo/assets/59613220/796a72ae-c7c5-49b9-866f-add32faab07c)
![2 request_in_xml](https://github.com/DefectDojo/django-DefectDojo/assets/59613220/1413eba8-6c4c-465e-aadf-ad472a2b4856)
![3 Steps_to_reproduce_empty](https://github.com/DefectDojo/django-DefectDojo/assets/59613220/80ef1a12-6d19-4860-b916-19e6dc965bf4)


Note that I only replace the request and response fields in the XML, the rest of the file is as in the github page, that's why the severity of the find is incorrect and so on.
[qualys_webapp_one_vuln_modified.zip](https://github.com/DefectDojo/django-DefectDojo/files/15402153/qualys_webapp_one_vuln_modified.zip)


Thanks in advance :)

**Reply by manuel-sommer** [reply]
I made a PR @Skuller99Dev for this issue. https://github.com/DefectDojo/django-DefectDojo/pull/10422

**Reply by manuel-sommer** [reply]
@mtesauro, could you mark my comment  above as the answer as I fixed it in a PR?

**Reply by mtesauro** [reply]
Oddly, GH won't let me.

![image](https://github.com/DefectDojo/django-DefectDojo/assets/3422419/99fa84c2-3cdc-4928-899d-c16a48480989)

**Reply by manuel-sommer** [reply]
weird.

**Reply by mtesauro** [reply]
Agreed

**Reply by manuel-sommer** [reply]
Ah, I guess it is because it is not from category  "Q&A", but rather from Category "General". https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-categories-for-discussions#default-categories

The question is if you can switch the category. From my point of view it makes sense to "close" / "mark as answered" old topics / issues in order to have easier tracking on relevant things. But I will stick with your decision.
